src/main/java/chopchop/parser/CommandParser.java

```diff
@@ -0,0 +1,102 @@
+// CommandParser.java
+
+package chopchop.parser;
+
+import java.util.Map;
+import java.util.List;
+import java.util.HashMap;
+import java.util.Optional;
+import java.util.ArrayList;
+
+import chopchop.util.Result;
+import chopchop.util.StringView;
+
+public class CommandParser {
+
+    private Result<Map<String, List<String>>> parseNamedArguments(StringView input) {
+
+        var ret = new HashMap<String, List<String>>();
+        while (input.size() > 0) {
+
+            if (input.find('/') != 0) {
+                break;
+            }
+
+            var currentArg = input.drop(1).bisect('/', input);
```

Comment:
I think this modifies the argument, which seems undesirable.

Reply:
yes, it is designed to modify its input arguments. this wouldn't be necessary if java had some semblance of decent features after 14 versions, like pattern matching;

this is how i use bisect in a proper language:
```cpp
std::string_view x;
std::string_view xs;

std::tie(x, xs) = bisect(input, ' ');
if(x.empty()) return { };

if(x[0] == '@')
{
	parse_tags(msg, x);
	std::tie(x, xs) = util::bisect(xs, ' ');
}

if(x[0] == ':')
{
	parse_prefix(msg, x);
	std::tie(x, xs) = util::bisect(xs, ' ');
}
```

unfortunately this can't be done in java (there's no `std::tie` equivalent, nor can there ever be). rather than introduce loads of syntactic noise by needing to (a) declare a pair to store the bisectionr result; (b) declare or assign a variable for the first part; (c) declare or assign a variable for the second part, it is more convenient this way.

besides, `bisect()` clearly documents its mutating behaviour.